components/tools/OmeroWeb/omeroweb/api/api_exceptions.py

```diff
@@ -50,6 +50,22 @@ def __init__(self, message, stacktrace=None):
         self.stacktrace = stacktrace
 
 
+class MethodNotSupportedError(Exception):
+    """
+    An exception that will result in a response status of 405.
+
+    Raised if user tries to DELETE, POST or PUT for an Object
+    where we don't support those methods.
+    """
+
+    status = 405
+
+    def __init__(self, message, stacktrace=None):
```

Comment:
I'm not seeing any use of the `stacktrace` keyword argument in the implementation. Is it your intention to use it for something? If so, what? If you want to have a unified constructor prototype that is similar to the other exceptions you have above then I would suggest making a class hierarchy so you are not having to implement, and document, this every time.